plugin/forward/forward.go

```diff
@@ -68,6 +72,14 @@ func (f *Forward) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg
 		return plugin.NextOrFailure(f.Name(), f.Next, ctx, w, r)
 	}
 
+	if f.maxQueryCount > 0 {
+		count := atomic.AddInt64(&(f.queryCount), 1)
+		defer atomic.AddInt64(&(f.queryCount), -1)
+		if count > f.maxQueryCount {
+			return dns.RcodeServerFailure, errors.New("inflight forward queries exceeded maximum")
```

Comment:
No needs to construct the error object here on each query, it can be constructed once as a global var, see `ErrNoHealthy` for example

Reply:
Thanks, fixed.